dot/types/grandpa_test.go

```diff
@@ -7,55 +7,161 @@ import (
 		Number: 999,
 	}
+	secondVote := GrandpaVote{
+		Hash:   common.Hash{0xd, 0xc, 0xb, 0xa},
+		Number: 999,
+	}
+	signature := [64]byte{1, 2, 3, 4}
+	keypair, err := ed25519.GenerateKeypair()
+	require.NoError(t, err)
+
+	var authorityID [32]byte
+	copy(authorityID[:], keypair.Public().Encode())
+
+	equivocation := GrandpaEquivocation{
+		RoundNumber:     0,
+		ID:              authorityID,
+		FirstVote:       firstVote,
+		FirstSignature:  signature,
+		SecondVote:      secondVote,
+		SecondSignature: signature,
+	}
 
-	enc, err := scale.Marshal(testVote)
+	equivPreVote := PreVoteEquivocation(equivocation)
+	equivVote := NewGrandpaEquivocation()
+	err = equivVote.Set(equivPreVote)
 	require.NoError(t, err)
-	require.Equal(t, exp, enc)
+	encoding := scale.MustMarshal(*equivVote)
 
-	dec := GrandpaVote{}
-	err = scale.Unmarshal(enc, &dec)
+	grandpaEquivocation := NewGrandpaEquivocation()
+	err = scale.Unmarshal(encoding, grandpaEquivocation)
 	require.NoError(t, err)
-	require.Equal(t, testVote, dec)
+	require.Equal(t, equivVote, grandpaEquivocation)
+}
 
+func Test_GrandpaVote(t *testing.T) {
+	t.Parallel()
+	expectedEncoding := common.MustHexToBytes("0x0a0b0c0d00000000000000000000000000000000000000000000000000000000e7030000")
+	vote := GrandpaVote{
+		Hash:   common.Hash{0xa, 0xb, 0xc, 0xd},
+		Number: 999,
+	}
+
+	encoding := scale.MustMarshal(vote)
+	require.Equal(t, expectedEncoding, encoding)
+
+	grandpaVote := GrandpaVote{}
+	err := scale.Unmarshal(encoding, &grandpaVote)
+	require.NoError(t, err)
+	require.Equal(t, vote, grandpaVote)
 }
 
 func TestEncodeSignedVote(t *testing.T) {
-	exp := common.MustHexToBytes("0x0a0b0c0d00000000000000000000000000000000000000000000000000000000e7030000010203040000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000506070800000000000000000000000000000000000000000000000000000000") //nolint:lll
-	var testVote = GrandpaVote{
+	t.Parallel()
+	expectedEncoding := common.MustHexToBytes("0x0a0b0c0d00000000000000000000000000000000000000000000000000000000e7030000010203040000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000506070800000000000000000000000000000000000000000000000000000000") //nolint:lll
+	vote := GrandpaVote{
 		Hash:   common.Hash{0xa, 0xb, 0xc, 0xd},
 		Number: 999,
 	}
-	var testSignature = [64]byte{1, 2, 3, 4}
-	var testAuthorityID = [32]byte{5, 6, 7, 8}
+	signature := [64]byte{1, 2, 3, 4}
+	authorityID := [32]byte{5, 6, 7, 8}
 
-	sv := GrandpaSignedVote{
-		Vote:        testVote,
-		Signature:   testSignature,
-		AuthorityID: testAuthorityID,
+	signedVote := GrandpaSignedVote{
+		Vote:        vote,
+		Signature:   signature,
+		AuthorityID: authorityID,
 	}
-	enc, err := scale.Marshal(sv)
-	require.NoError(t, err)
-	require.Equal(t, exp, enc)
+	encoding := scale.MustMarshal(signedVote)
+	require.Equal(t, expectedEncoding, encoding)
```

Comment:
All this is unnecessary renaming. Please change it back.

Reply:
Why is it unnecessary? I think we opt to use := over var when we can in our code (correct time if I am wrong) so I changed that, and then sv and enc are vague so I made them more clear